prov/gni/src/gnix_rma.c

```diff
@@ -740,6 +794,140 @@ int _gnix_rma_post_rdma_chain_req(void *data)
 	return FI_SUCCESS;
 }
 
+static void __gnix_rma_more_fill_pd(struct gnix_fab_req *req,
+				    struct gnix_tx_descriptor *txd)
+{
+	gni_ct_put_post_descriptor_t *more_put;
+	gni_ct_get_post_descriptor_t *more_get;
+	gni_mem_handle_t mdh;
+	struct gnix_fab_req *more_req;
+	struct slist_entry *current;
+	int entries = 0, idx = 0;
+
+	current = req->rma.sle.next; /* First sub descriptor slist entry */
+	while (current != NULL) {
+		entries++;
+		current = current->next;
+	}
+
+	GNIX_INFO(FI_LOG_EP_DATA, "MORE: %d sub descs to be populated\n", entries);
+
+	if (req->type == GNIX_FAB_RQ_RDMA_WRITE) {
+		txd->gni_more_ct_descs = malloc(entries*sizeof(gni_ct_put_post_descriptor_t));
+		more_put = (gni_ct_put_post_descriptor_t *)txd->gni_more_ct_descs;
+	}
+	else {
+		txd->gni_more_ct_descs = malloc(entries*sizeof(gni_ct_get_post_descriptor_t));
+		more_get = (gni_ct_get_post_descriptor_t *)txd->gni_more_ct_descs;
+	}
+
+	current = req->rma.sle.next;
+	while (current != NULL) {
+		/* Get fab_req pointer */
+		more_req = container_of(current, struct gnix_fab_req, rma.sle);
+
+		/* Populate txd based on type */
+		if (req->type == GNIX_FAB_RQ_RDMA_WRITE) {
+			more_put[idx].ep_hndl = more_req->vc->gni_ep;
+			more_put[idx].length = more_req->rma.len;
+			more_put[idx].remote_addr = more_req->rma.rem_addr;
+			more_put[idx].local_addr = (uint64_t)txd->int_buf;
+
+			_gnix_convert_key_to_mhdl_no_crc(
+					(gnix_mr_key_t *)&more_req->rma.rem_mr_key,
+					&mdh);
+			more_put[idx].remote_mem_hndl = mdh;
+
+			if (idx < entries)
+				more_put[idx].next_descr = &more_put[idx + 1];
+			else
+				more_put[idx].next_descr = NULL;
+		}
+		else {
+			/* TODO */
+			if (idx < entries)
+				more_get[idx].next_descr = &more_get[idx + 1];
+			else
+				more_get[idx].next_descr = NULL;
+		}
+		idx++;
+		current = current->next;
+	}
+}
+
+int _gnix_rma_more_post_req(void *data)
+{
+	struct gnix_fab_req *fab_req = (struct gnix_fab_req *)data;
+	struct gnix_fid_ep *ep = fab_req->gnix_ep;
+	struct gnix_nic *nic = ep->nic;
+	struct gnix_fid_mem_desc *loc_md;
+	struct gnix_tx_descriptor *txd;
+	gni_mem_handle_t mdh;
+	gni_return_t status;
+	int rc;
+
+	GNIX_INFO(FI_LOG_EP_DATA, "MORE: Entered work_fn\n");
```

Comment:
Okay, I plan on removing most of my GNIX_INFO Calls, or as you suggested replacing them with GNIX_DEBUG().